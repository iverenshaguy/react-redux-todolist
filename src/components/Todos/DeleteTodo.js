import React from 'react';
import PropTypes from 'prop-types';
import Button from '../shared/Button';

const DeleteTodo = props => {
  return (
    <Button
      buttonType="button"
      btnClass="close"
      buttonText="&times;"
      onClick={props.onClick}
    />
  );
};

DeleteTodo.proptypes = {
  onClick: PropTypes.func.isRequired
};

export default DeleteTodo;
