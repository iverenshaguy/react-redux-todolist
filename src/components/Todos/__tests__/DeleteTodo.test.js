import DeleteTodo from '../DeleteTodo';

describe('DeleteTodo', () => {
  const clickFunction = jest.fn();

  it('renders correctly', () => {
    const wrapper = shallow(<DeleteTodo onClick={clickFunction} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('mounts correctly', () => {
    const wrapper = mount(<DeleteTodo onClick={clickFunction} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('calls a function on click', () => {
    const wrapper = shallow(<DeleteTodo onClick={clickFunction} />);

    wrapper.simulate('click');
    expect(clickFunction).toHaveBeenCalled();
  });
});
