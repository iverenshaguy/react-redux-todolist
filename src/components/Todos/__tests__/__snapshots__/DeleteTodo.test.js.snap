// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteTodo mounts correctly 1`] = `
<DeleteTodo
  onClick={[Function]}
>
  <Button
    btnClass="close"
    buttonText="×"
    buttonType="button"
    onClick={[Function]}
  >
    <button
      className="close"
      onClick={[Function]}
      type="button"
    >
      ×
    </button>
  </Button>
</DeleteTodo>
`;

exports[`DeleteTodo renders correctly 1`] = `
<Button
  btnClass="close"
  buttonText="×"
  buttonType="button"
  onClick={[Function]}
/>
`;
